proposals/scopes.md

```diff
@@ -88,6 +92,25 @@ Also, such post-processing could include hiding unnecessary properties from obje
 
 ## Detailed design
 
+The sourcemap should include information for every scope in the generated source and every scope in the original sources that contains code which appears in the generated source.
+More precisely, for every location `loc_gen` in the generated code that is mapped to `loc_orig` in the original code:
+- the generated scopes described in the sourcemap which contain `loc_gen` should be exactly the scopes in the generated source which contain `loc_gen`
+- the original scopes described in the sourcemap which contain `loc_gen` should be exactly
+  - the scopes in the original source which contain `loc_orig` and
+  - if `loc_gen` is in an inlined function, the scopes in the original source which contain the function call that was inlined
+
+The following information describes a scope in the source map:
+- the type of the scope (e.g. block, function or module scope)
+- whether this scope appears in the original and/or the generated source
+- the start and end locations of the scope in the generated source
+- only for scopes that appear in the original source: the start and end locations of the scope in the original source
```

Comment:
- I thought I needed the scope's original start/end locations in case of function inlining. When I worked out the algorithm for computing original frames and scopes (in [this commit](https://github.com/hbenl/tc39-proposal-scope-mapping/commit/df6199e12117de4e47944890d569cb245af19d37)), I realized that I need the callsite (the location from where the inlined function was called) instead. I've updated the description accordingly [here](https://github.com/hbenl/source-map-rfc/commit/d6cef06fecae2c86a81551f217faf93f47efabf5)
- locations (both original and generated) could be represented by indices into the mappings array (which could reduce the sourcemap size) but we could lose some accuracy because not every location has a mapping. I'm not yet working on the question how the scope information should be encoded - first I want to work out the algorithm for merging the sourcemaps of two consecutive transpilation steps to figure out which scope information will be needed and how important the accuracy of the locations is (my hunch is that accurate locations could be important to facilitate merging).